src/werkzeug/formparser.py

```diff
@@ -461,6 +469,160 @@ def validate_boundary(self, boundary):
+
+        def _state_headers(self, line):
+            while True:
+                if line is None:
+                    self.parent.fail("unexpected end of file during headers")
+
+                line = _to_str(line)
+                line, line_terminated = _line_parse(line)
+                if not line_terminated:
+                    self.parent.fail("unexpected end of line in multipart header")
+
+                if not line:
+                    self._headers = Headers(self._headers)
+                    if self._headers_only:
+                        yield self._headers, ParseEnums.done
+                    else:
+                        disposition = self._headers.get("content-disposition", None)
+                        if disposition is None:
+                            self.parent.fail("missing Content-Disposition header")
+                        else:
+                            disposition, extra = parse_options_header(disposition)
+                            transfer_encoding = self.parent.get_part_encoding(
+                                self._headers
+                            )
+                            if transfer_encoding is not None:
+                                if transfer_encoding == "base64":
+                                    transfer_encoding += "_codec"
+
+                            try:
+                                self._codec = codecs.lookup(transfer_encoding)
+                            except Exception:
+                                self.parent.fail(
+                                    f"cannot decode transfer-encoding: "
+                                    f"{transfer_encoding}"
+                                )
+
+                            self._name = extra.get("name")
+                            self._filename = extra.get("filename")
+
+                            if self._filename is not None:
+                                yield (
+                                    _begin_file,
+                                    (self._headers, self._name, self._filename),
+                                ), ParseEnums.output
+                            else:
+                                yield (
+                                    _begin_form,
+                                    (self._headers, self._name),
+                                ), ParseEnums.output
+                elif line[0] in "\t" and self._headers:
+                    key, value = self._headers[-1]
+                    self._headers[-1] = (key, value + "\n" + line[1:])
+                else:
+                    parts = line.split(":", 1)
+                    if len(parts) == 2:
+                        self._headers.append((parts[0].strip(), parts[1].strip()))
+
+                    yield None, ParseEnums.headers
+
+        def _state_output(self, line):
+            while True:
+                if not line:
+                    self.parent.fail("Unexpected end of file")
+                sline = line.rstrip()
```

Comment:
Nitpick: `sline` is kind of an inscrutable variable name. Please make it more informative.